src/promptflow/promptflow/_sdk/entities/_connection.py

```diff
@@ -763,7 +768,8 @@ def _get_custom_keys(cls, data: Dict):
                 not data["configs"][CustomStrongTypeConnectionConfigs.PROMPTFLOW_TYPE_KEY]
                 or not data["configs"][CustomStrongTypeConnectionConfigs.PROMPTFLOW_MODULE_KEY]
             ):
-                raise ValueError("custom_type and module are required for custom strong type connections.")
+                error = ValueError("custom_type and module are required for custom strong type connections.")
+                raise UserErrorException(message=str(error), error=error)
```

Comment:
why wrap?

Reply:
Based on offline discussions, we will further optimize the details and incorporate them into this PR.